ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/PutJobManagement_Test.java

```diff
@@ -0,0 +1,600 @@
+        
+        try {
+            final List<Ds3Object> objectsOne = Lists.newArrayList(new Ds3Object("testOne", 2));
+
+            HELPERS.startWriteJob(BUCKET_NAME, Lists.newArrayList(new Ds3Object("testOne", 2)));
+
+            final List<Ds3Object> objectsTwo = Lists.newArrayList(new Ds3Object("testTwo", 2));
+
+            HELPERS.startWriteJob(BUCKET_NAME, objectsTwo);
+
+            final CancelAllJobsSpectraS3Response response = client
+                    .cancelAllJobsSpectraS3(new CancelAllJobsSpectraS3Request());
+
+            assertTrue(client.getActiveJobsSpectraS3(new GetActiveJobsSpectraS3Request())
+                    .getActiveJobListResult().getActiveJobs().isEmpty());
+        } finally {
+            deleteAllContents(client, BUCKET_NAME);
+        }
+    }
+
+    @Test
+    public void truncateCancelAllJobsWithoutForce() throws IOException, SignatureException, XmlProcessingException, InterruptedException, URISyntaxException {
+
+        final int testTimeOutSeconds = 5;
+        final String book1 = "beowulf.txt";
+        final Path objPath1 = ResourceUtils.loadFileResource(RESOURCE_BASE_NAME + book1);
+        final String book2 = "ulysses.txt";
+        final Path objPath2 = ResourceUtils.loadFileResource(RESOURCE_BASE_NAME + book2);
+        final Ds3Object obj1 = new Ds3Object(book1, Files.size(objPath1));
+        final Ds3Object obj2 = new Ds3Object("place_holder_1", 5000000);
+        final Ds3Object obj3 = new Ds3Object(book2, Files.size(objPath2));
+        final Ds3Object obj4 = new Ds3Object("place_holder_2", 5000000);
+
+        try {
+            final Ds3ClientHelpers.Job putJob1 = HELPERS.startWriteJob(BUCKET_NAME, Lists
+                    .newArrayList(obj1, obj2));
+            final UUID jobId1 = putJob1.getJobId();
+            final SeekableByteChannel book1Channel = new ResourceObjectPutter(RESOURCE_BASE_NAME).buildChannel(book1);
+            client.putObject(new PutObjectRequest(BUCKET_NAME, book1, book1Channel, jobId1, 0, Files.size(objPath1)));
+
+            final Ds3ClientHelpers.Job putJob2 = HELPERS.startWriteJob(BUCKET_NAME, Lists
+                    .newArrayList(obj3, obj4));
+            final UUID jobId2 = putJob2.getJobId();
+            final SeekableByteChannel book2Channel = new ResourceObjectPutter(RESOURCE_BASE_NAME).buildChannel(book2);
+            client.putObject(new PutObjectRequest(BUCKET_NAME, book2, book2Channel, jobId2, 0, Files.size(objPath2)));
+
+            final Ds3ClientHelpers.Job putJob3 = HELPERS.startWriteJob(BUCKET_NAME, Lists
+                    .newArrayList(new Ds3Object("place_holder_3", 1000000)));
+            final UUID jobId3 = putJob3.getJobId();
+
+            //make sure black pearl has updated the first 2 jobs to show 1 object in cache each
+            final long startTime = System.nanoTime();
+            boolean cachedSizeUpdated = false;
+            while (!cachedSizeUpdated) {
```

Comment:
I've seen this block of code repeated a few times as well.  Can this be pulled out into it's own function?